cli/metrics/client.go

```diff
@@ -32,10 +32,11 @@ type client struct {
 
 // Command is a command
 type Command struct {
-	Command string `json:"command"`
-	Context string `json:"context"`
-	Source  string `json:"source"`
-	Status  string `json:"status"`
+	Command  string `json:"command"`
+	Context  string `json:"context"`
+	Source   string `json:"source"`
+	Status   string `json:"status"`
+	Metadata []byte `json:"metadata,omitempty"`
```

Comment:
Should this not be
```suggestion
	Metadata map[string]interface{} `json:"metadata,omitempty"`
```
if it’s JSON?

Reply:
I think it's better to send plain bytes so server doesn't care about a specific type. After all it's just JSON.

Reply:
With bytes it’s going to write an unreadable blob, is this really what we want?
If the server is in JS for instance that’s not going to be very handy.
Even if it’s in Go it’s likely much easier to unserialize JSON all the way?

Reply:
E.g. https://play.golang.org/p/EQWj-1dC7it

Reply:
Looking at your comment https://github.com/docker/compose-cli/pull/2104#discussion_r751300156 we could use a map if you just want to put specific field to the heartbeat.
Or plain json as a string field instead of a slice of bytes so we push everything in `Metadata` to the heartbeat without bothering decoding its content. I think that's what the `Features` one does today.